src/main.ts

```diff
@@ -78,9 +78,25 @@ export default class TesseractOcrPlugin extends Plugin {
 			if(this.isMarkdown(file.name)) {
 				checkedFilesCounter++;
 
-				let linkRegex = /!\[\[.*\]\](?!<details>)/g
-				let content = await this.app.vault.cachedRead(file);
-				let newContent = content;
+		        let content = await this.app.vault.cachedRead(file);
+		        let newContent = content;
+
+		        // Define regex for relative legacy links that contain the linkPath
+				let regexPattern = `\\[.*?\\]\\((.*?)(${this.settings.imagePath}.*?\\..*?)\\)`;
+				let legacyLinkRegex = new RegExp(regexPattern, 'g'); // 'g' for global replacement
+
+				// Convert legacy links to absolute wiki style links
+				newContent = newContent.replace(legacyLinkRegex, (match, beforePath, imagePath) => {
+				    if (this.settings.debug == true) console.log(match, beforePath, imagePath);
+				    // Use the imagePath directly since it already contains the path from this.settings.imagePath onwards
+				    return `[[${unescape(imagePath)}]]`;
+				});
```

Comment:
A legacy style backlink is a link in the standard markdown format e.g.: ![Link to my page](../uploads/image.png)

Before obsidian, I had a few markdown notes apps that were creating links to embdded images in this way, and since Obsidian handled it gracefully, it has been a non issue, and I have yet to migrate the link format.

When i saw this plugin was only operating on the standard backlink style format, I took it as an opportunity to write a bit of code to clean up the old format.

I have already run this chunk against my vault, and I definitely don't feel strongly if this snippet makes it back in to the PR.

How about I break it out to a separate branch and we can move forward with the rest of the changes? I will make the tweaks and address the rest of your feedback when I get some time this weekend.